client/src/nv_ingest_client/primitives/jobs/job_spec.py

```diff
@@ -152,3 +160,208 @@ def add_task(self, task) -> None:
+        A dictionary that maps document types to a list of `JobSpec` instances.
+    """
+
+    def __init__(self, job_specs_or_files: Optional[Union[List[JobSpec], List[str]]] = None) -> None:
+        """
+        Initializes the BatchJobSpec instance.
+
+        Parameters
+        ----------
+        job_specs_or_files : Optional[Union[List[JobSpec], List[str]]], optional
+            A list of either `JobSpec` instances or file paths. If provided, the
+            instance will be initialized with the corresponding job specifications.
+        """
+        self._file_type_to_job_spec = defaultdict(list)
+
+        if job_specs_or_files:
+            if isinstance(job_specs_or_files[0], JobSpec):
+                self.from_job_specs(job_specs_or_files)
+            elif isinstance(job_specs_or_files[0], str):
+                self.from_files(job_specs_or_files)
+            else:
+                raise ValueError("Invalid input type for job_specs. Must be a list of JobSpec or file paths.")
+
+    def from_job_specs(self, job_specs: Union[JobSpec, List[JobSpec]]) -> None:
+        """
+        Initializes the batch with a list of `JobSpec` instances.
+
+        Parameters
+        ----------
+        job_specs : Union[JobSpec, List[JobSpec]]
+            A single `JobSpec` or a list of `JobSpec` instances to add to the batch.
+        """
+        if isinstance(job_specs, JobSpec):
+            job_specs = [job_specs]
+
+        for job_spec in job_specs:
+            self.add_job_spec(job_spec)
+
+    def from_files(self, files: Union[str, List[str]]) -> None:
+        """
+        Initializes the batch by generating job specifications from file paths.
+
+        Parameters
+        ----------
+        files : Union[str, List[str]]
+            A single file path or a list of file paths to create job specifications from.
+        """
+        from nv_ingest_client.util.util import create_job_specs_for_batch
+        from nv_ingest_client.util.util import generate_matching_files
+
+        if isinstance(files, str):
+            files = [files]
+
+        matching_files = list(generate_matching_files(files))
+        if not matching_files:
+            logger.warning(f"No files found matching {files}.")
+            return
+
+        job_specs = create_job_specs_for_batch(matching_files)
+        for job_spec in job_specs:
+            self.add_job_spec(job_spec)
+
+    def _from_dataset(self, dataset: str, shuffle_dataset: bool = True) -> None:
+        """
+        Internal method to initialize the batch from a dataset.
+
+        Parameters
+        ----------
+        dataset : str
+            The path to the dataset file.
+        shuffle_dataset : bool, optional
+            Whether to shuffle the dataset files before adding them to the batch, by default True.
+        """
+        with open(dataset, "rb") as file:
+            dataset_bytes = BytesIO(file.read())
+
+        logger.debug(get_dataset_statistics(dataset_bytes))
```

Comment:
get_dataset_statistics can be a really heavy-weight call depending on how many files are in the dataset, and even if our log level is below debug, it will still be called and then discarded. This work around looks ugly, but I think it makes sense in this case.

```suggestion
if logger.isEnabledFor(logging.DEBUG):
    logger.debug(get_dataset_statistics(dataset_bytes))
```